compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/IfNode.java

```diff
@@ -893,14 +893,28 @@ private ValueNode canonicalizeConditionalViaImplies(ValueNode trueValue, ValueNo
         return null;
     }
 
+    private boolean isEmptyBlock(ValueNode value, AbstractBeginNode successor) {
```

Comment:
Can we call this isSafeConditionalInput?  And some comments explaining the thinking.  Basically Constants and Parameters are cheap to materialize and if we have a schedule then an empty block means the value is already materialized so we aren't adding unconditional work by converting to an conditional form.